src/icmp/Icmp4.cc

```diff
@@ -115,9 +114,9 @@ Icmp4::SendEcho(Ip::Address &to, int opcode, const char *payload, int len)
     // Construct ICMP packet data content
     echo = reinterpret_cast<icmpEchoData *>(reinterpret_cast<char *>(pkt) + sizeof(*icmp));
     echo->opcode = (unsigned char) opcode;
-    memcpy(&echo->tv, &current_time, sizeof(struct timeval));
+    echo->tv = std::chrono::system_clock::now();
```

Comment:
Official memcpy() code does not raise alignment concerns because it only uses the address of `echo->tv`. The new code does raise those concerns because it now treats memory in the middle of some I/O buffer as usable for time_point storage (an STL type with unknown (to us) alignment requirements).